pipeline/foraging_model.py

```diff
@@ -0,0 +1,301 @@
+    aic: float   # AIC
+    bic: float   # BIC
+    lpt: float       # Likelihood-Per-Trial raw
+    lpt_aic: float   # Likelihood-Per-Trial with AIC penalty 
+    lpt_bic: float   # Likelihood-Per-Trial with AIC penalty 
+    prediction_accuracy: float    # non-cross-validated prediction accuracy 
+    cross_valid_accuracy_fit: float     # cross-validated accuracy (fitting set)
+    cross_valid_accuracy_test: float    # cross-validated accuracy (testing set)
+    cross_valid_accuracy_test_bias_only = NULL: float    # accuracy predicted only by bias (testing set)
+    """
+
+    key_source = (foraging_analysis.SessionTaskProtocol() & 'session_task_protocol = 100' & 'session_real_foraging'
+                  ) * Model() #& (experiment.Session & 'session_date > "2021-01-01"')
+
+    class FittedParam(dj.Part):
+        definition = """
+        -> master
+        -> Model.Param
+        ---
+        fitted_value: float
+        """
+    
+    class PredictiveChoiceProb(dj.Part):
+        definition = """
+        # Could be used to compute latent value Q (for models that have Q). Ignored trial skipped.
+        -> master
+        -> experiment.SessionTrial
+        -> experiment.WaterPort
+        ---
+        choice_prob: float        
+        """
+
+    def make(self, key):
+        choice_history, reward_history, p_reward, q_choice_outcome = get_session_history(key)
+        model_str = (Model & key).fetch('fit_cmd')
+
+        # --- Actual fitting ---
+        model_comparison_this = BanditModelComparison(choice_history, reward_history, model=model_str)
+        model_comparison_this.fit(pool='', plot_predictive=None, if_verbose=False)  # Parallel on sessions, not on DE
+        model_comparison_this.cross_validate(pool='', k_fold=2, if_verbose=False)
+
+        # ------ Grab results ----
+        fit_result = model_comparison_this.results_raw[0]
+        cross_valid_result = model_comparison_this.prediction_accuracy_CV.iloc[0]
+
+        # Insert session fitted stats
+        self.insert1(dict(**key,
+                          n_trials=fit_result.n_trials,
+                          n_params=fit_result.k_model,
+                          log_likelihood=fit_result.log_likelihood,
+                          aic=fit_result.AIC,
+                          bic=fit_result.BIC,
+                          lpt=fit_result.LPT,
+                          lpt_aic=fit_result.LPT_AIC,
+                          lpt_bic=fit_result.LPT_BIC,
+                          prediction_accuracy=fit_result.prediction_accuracy,
+                          cross_valid_accuracy_fit=np.mean(cross_valid_result.prediction_accuracy_fit),
+                          cross_valid_accuracy_test=np.mean(cross_valid_result.prediction_accuracy_test),
+                          cross_valid_accuracy_test_bias_only=np.mean(cross_valid_result.prediction_accuracy_test_bias_only),
+                          ),
+                     skip_duplicates=True, allow_direct_insert=True)
```

Comment:
You probably don't need `allow_direct_insert=True` here (it is only needed if you are inserting entries to Imported/Computed tables from outside of the `make()`

Reply:
Also, why the need for `skip_duplicates=True` here? Do you expect somehow through the `populate()` there will be duplicates in the fitted session model?

Reply:
Both removed -- I somehow needed them when I debugged the code...